rust-driver/src/lib.rs

```diff
@@ -0,0 +1,157 @@
+    clippy::indexing_slicing,
+    clippy::inline_asm_x86_intel_syntax,
+    clippy::arithmetic_side_effects,
+ 
+    // clippy::pattern_type_mismatch, // cause some false postive and unneeded copy
+    clippy::print_stderr,
+    clippy::print_stdout,
+    clippy::rc_buffer,
+    clippy::rc_mutex,
+    clippy::rest_pat_in_fully_bound_structs,
+    clippy::same_name_method,
+    clippy::self_named_module_files,
+    // clippy::shadow_reuse, it’s a common pattern in Rust code
+    // clippy::shadow_same, it’s a common pattern in Rust code
+    clippy::shadow_unrelated,
+    clippy::str_to_string,
+    clippy::string_add,
+    clippy::string_to_string,
+    clippy::unimplemented,
+    clippy::unnecessary_self_imports,
+    clippy::unneeded_field_pattern,
+    // clippy::unreachable, // the unreachable code should unreachable otherwise it's a bug
+    clippy::unwrap_in_result,
+    clippy::unwrap_used, 
+    clippy::use_debug,
+    clippy::verbose_file_reads,
+    clippy::wildcard_enum_match_arm,
+
+    // // The followings are selected lints from 1.61.0 to 1.67.1
+    clippy::as_ptr_cast_mut,
+    clippy::derive_partial_eq_without_eq,
+    clippy::empty_drop,
+    clippy::empty_structs_with_brackets,
+    clippy::format_push_string,
+    clippy::iter_on_empty_collections,
+    clippy::iter_on_single_items,
+    clippy::large_include_file,
+    clippy::manual_clamp,
+    clippy::suspicious_xor_used_as_pow,
+    clippy::unnecessary_safety_comment,
+    clippy::unnecessary_safety_doc,
+    clippy::unused_peekable,
+    clippy::unused_rounding,    
+
+    // The followings are selected restriction lints from rust 1.68.0 to 1.71.0
+    // clippy::allow_attributes, still unstable
+    clippy::impl_trait_in_params,
+    clippy::let_underscore_untyped,
+    clippy::missing_assert_message,
+    clippy::multiple_unsafe_ops_per_block,
+    clippy::semicolon_inside_block,
+    // clippy::semicolon_outside_block, already used `semicolon_inside_block`
+    clippy::tests_outside_test_module,
+    // 1.71.0
+    clippy::default_constructed_unit_structs,
+    clippy::items_after_test_module,
+    clippy::manual_next_back,
+    clippy::manual_while_let_some,
+    clippy::needless_bool_assign,
+    clippy::non_minimal_cfg,
+)]
+
+#![cfg_attr(
+    test,
+    allow(
+        clippy::indexing_slicing,
+        unused_results,
+        clippy::unwrap_used,
+        clippy::unwrap_in_result,
+        clippy::expect_used,
+        clippy::as_conversions,
+        clippy::shadow_unrelated,
+        clippy::arithmetic_side_effects,
+        clippy::let_underscore_untyped,
+        clippy::pedantic, 
+        clippy::default_numeric_fallback,
+        clippy::print_stderr,
+        unsafe_code,
+    )
+)]
+
+mod rdma;
+
+/// adaptor device: hardware, software, emulated
+mod device;
+
+/// some useful types 
+pub mod types;
+
+/// A recycle buffer allocator for ack buffer
+mod recycle_allocator;
+
+/// utility functions
+mod utils;
+
+pub use device::PAGE_SIZE;
```

Comment:
The original purpose of `PAGE_SIZE` was to inform users about the page size used by the hardware. Since we may have different `PAGE_SIZE` for different cards, I will remove it and provide an interface in device adaptor later.